[IMP] inventory: troubleshooting 3rd party carrier (#8788)
odoo/documentation: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/third_party_shipper.rst

addresses, :doc:`automatically calculate shipping costs <delivery_method>`, and :doc:`generate
shipping labels <../advanced_operations_shipping/labels>`.

> In Odoo, shipping carriers can be applied to sales orders (SO), invoices, or delivery orders. For

Reviewer: ```suggestion
In Odoo, shipping carriers can be applied to sales orders (SOs), invoices, or delivery orders. For
```
If it's multiple sales orders, should we add an -s here? I know it looks a bit weird so it's up to you

delivery to specific addresses, :doc:`automatically calculate shipping costs <delivery_method>`, and
:doc:`generate shipping labels <../advanced_operations_shipping/labels>`.

In Odoo, shipping carriers can be applied to a sales order (SO), invoice, or delivery order. For

> tips on resolving common issues when configuring shipping connectors, :ref:`skip to this section
> <inventory/shipping_receiving/third-party-troubles>`.

Reviewer 2: Adding a little more meaning into the link label, so it is clear to the user where they will end up after clicking the link :)
```suggestion
tips on resolving common issues when configuring shipping connectors, skip to the :ref:`Troubleshooting
<inventory/shipping_receiving/third-party-troubles>`  section.
```

Author: This is really good advice! I'll make sure all the link labels are named, from now on!

> tips on resolving common issues when configuring shipping connectors, :ref:`skip to this section
> <inventory/shipping_receiving/third-party-troubles>`.

Reviewer 2: Nitpick about meaningful link names, just so its clear to the reader which section you are referring to :)
```suggestion
tips on resolving common issues when configuring shipping connectors, skip to the
:ref:`Troubleshooting <inventory/shipping_receiving/third-party-troubles>` section.
```

.. image:: third_party_shipper/delivery-info.png
   :align: center
   :alt: Show the delivery order's "Additional info" tab.

.. _inventory/shipping_receiving/third-party-troubles:

Troubleshooting
===============

Since shipping connectors are challenging to set up, here are some checks to try when things are
not working as expected:

#. Ensure the :ref:`warehouse information <inventory/shipping_receiving/configure-source-address>`

>    (e.g. address and phone number) in Odoo is correct **and** match the records saved in the

Reviewer: ```suggestion
   (e.g., address and phone number) in Odoo is correct **and** match the records saved in the
```

   shipping provider's website.
#. Verify that the :ref:`package type <inventory/warehouses_storage/package-type>` and parameters
   are valid for the shipping carrier. To check, ensure the shipment can be directly created on the
   shipping carrier's website.
#. When encountering a price mismatch between Odoo's estimated cost and the provider's charge, first
   ensure the delivery method is set to :ref:`production environment
   <inventory/shipping_receiving/production-env>`. Then, create the shipment in both the carrier's
   website and Odoo, and compare the prices in the *debug logs*.

Debug logs
----------

> To track inconsistencies shipping data across Odoo and the shipping carrier, activate the debug log.

Reviewer: ```suggestion
To track inconsistencies in the shipping data between Odoo and the shipping carrier, activate the debug log.
```

.. image:: third_party_shipper/delivery-info.png
   :align: center
   :alt: Show the delivery order's "Additional info" tab.

.. _inventory/shipping_receiving/third-party-troubles:

Troubleshooting
===============

Since shipping connectors can sometimes be complex to set up, here are some checks to try when
things are not working as expected:

#. Ensure the :ref:`warehouse information <inventory/shipping_receiving/configure-source-address>`

>    (e.g., address and phone number) in Odoo is correct **and** match the records saved in the

Reviewer 3: ```suggestion
   (e.g. address and phone number) in Odoo is correct **and** match the records saved in the
```

>    (e.g., address and phone number) in Odoo is correct **and** match the records saved in the

Reviewer 2: Should this be "matches"?
```suggestion
   (e.g., address and phone number) in Odoo is correct **and** matches the records saved in the
```

   shipping provider's website.
#. Verify that the :ref:`package type <inventory/warehouses_storage/package-type>` and parameters
   are valid for the shipping carrier. To check, ensure the shipment can be directly created on the
   shipping carrier's website.
#. When encountering a price mismatch between Odoo's estimated cost and the provider's charge, first
   ensure the delivery method is set to :ref:`production environment
   <inventory/shipping_receiving/production-env>`.

   Then, create the shipment in both the carrier's website and Odoo, and verify the prices are the
   same across Odoo, the shipping provider, and in the *debug logs*.

   .. example::
      When checking for a price mismatch in the debug logs, if the request says the package weighs

>       six kilograms, but the response from FedEx says the package weights seven kilograms, it is
>       concluded the issue is on FedEx's side.

Reviewer 3: ```suggestion
      six kilograms, but the response from FedEx says the package weights seven kilograms, it
      concludes that the issue is on FedEx's side.
```

Debug log
---------

Track shipping data inconsistencies by activating debug logging. To do that, go to the delivery
method's configuration page (:menuselection:`Inventory app --> Configuration --> Shipping
Method`), and select the desired shipping method. Click the :guilabel:`No Debugging` smart button to
activate :guilabel:`Debug Requests`.

.. image:: third_party_shipper/no-debug.png
   :align: center
   :alt: Show the "No Debug" smart button.

With :guilabel:`Debug Requests` activated, each time the shipping connector is used to estimate the
cost of shipping, records are saved in the :guilabel:`Logging` report. To access the report, turn on

> :ref:`developer mode <developer-mode>` and go to :menuselection:`Settings app --> Technical -->

Reviewer 3: ```suggestion
:ref:`developer mode <developer-mode>`, and go to :menuselection:`Settings app --> Technical -->
```

> Database Structure: Logging`.

Reviewer 3: ```suggestion
Database Structure section --> Logging`.
```

.. note::

>    Logs are created for a shipping method each time the :ref:`"Get Rate"

Reviewer 3: ```suggestion
   Logs are created for a shipping method each time the :ref:`Get Rate
```

   <inventory/shipping_receiving/third-party-rate>` button is clicked on :abbr:`SOs (Sales Orders)`
   and invoices, **and** when a customer adds the shipping carrier to their order through the
   *Website* app.

.. image:: third_party_shipper/log.png
   :align: center
   :alt: Show how to find the "Logging" option from the "Technical" menu.

> Click the *HTTP request* line item to open a detailed page and verify the correct information is

Reviewer 3: ```suggestion
Click the *HTTP request* line item to open a detailed page, and verify the correct information is
```
